crates/biome_string_case/src/lib.rs

```diff
@@ -415,25 +419,207 @@ const LEADING_BIT_INDEX_TO_CASE: [Case; 11] = [
+    /// See also <https://www.unicode.org/reports/tr35/tr35-collation.html#CLDR_Collation_Algorithm>.
+    const COLLATION: [u8; 128] = [
+        b'\0', 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x0e, 0x0f, 0x10, 0x11, 0x12, 0x13,
+        0x14, 0x15, 0x16, 0x17, 0x18, 0x19, 0x1a, 0x1b, 0x1c, 0x1d, 0x1e, 0x1f, 0x7f, b'\t', b'\n',
+        0x0b, 0x0c, b'\r', b' ', b'_', b'-', b',', b';', b':', b'!', b'?', b'.', b'\'', b'"', b'(',
+        b')', b'[', b']', b'{', b'}', b'@', b'*', b'/', b'\\', b'&', b'#', b'%', b'`', b'^', b'+',
+        b'<', b'=', b'>', b'|', b'~', b'$', b'0', b'1', b'2', b'3', b'4', b'5', b'6', b'7', b'8',
+        b'9', b'A', b'a', b'B', b'b', b'C', b'c', b'D', b'd', b'E', b'e', b'F', b'f', b'G', b'g',
+        b'H', b'h', b'I', b'i', b'J', b'j', b'K', b'k', b'L', b'l', b'M', b'm', b'N', b'n', b'O',
+        b'o', b'P', b'p', b'Q', b'q', b'R', b'r', b'S', b's', b'T', b't', b'U', b'u', b'V', b'v',
+        b'W', b'w', b'X', b'x', b'Y', b'y', b'Z', b'z',
+    ];
+}
+impl AsciiCollator for CldrAsciiCollator {
+    const WEIGHTS: [u8; 256] = ascii_collation_weight_from(&Self::COLLATION);
+}
+
+/// Generate the collation weight table from an ASCII collation table.
+/// The last 128 bytes are mapped to themselves.
+pub const fn ascii_collation_weight_from(collation_table: &[u8; 128]) -> [u8; 256] {
+    let mut result = [0u8; 256];
+    let mut i = 0;
+    while i < collation_table.len() {
+        debug_assert!(result[collation_table[i] as usize] == 0);
+        result[collation_table[i] as usize] = i as u8;
+        i += 1;
+    }
+    while i < result.len() {
+        result[i] = i as u8;
+        i += 1;
+    }
+    result
+}
+
 pub trait StrLikeExtension: ToOwned {
     /// Returns the same value as String::to_lowercase. The only difference
     /// is that this functions returns ```Cow``` and does not allocate
     /// if the string is already in lowercase.
-    fn to_ascii_lowercase_cow(&self) -> std::borrow::Cow<Self>;
+    fn to_ascii_lowercase_cow(&self) -> Cow<Self>;
+
+    /// Compare two strings using a natural ASCII order.
+    ///
+    /// Uppercase letters come first (e.g. `A` < `a` < `B` < `b`)
+    /// and number are compared in a human way (e.g. `9` < `10`).
+    fn ascii_nat_cmp(&self, other: &Self) -> Ordering;
 }
 
 pub trait StrOnlyExtension: ToOwned {
     /// Returns the same value as String::to_lowercase. The only difference
     /// is that this functions returns ```Cow``` and does not allocate
     /// if the string is already in lowercase.
-    fn to_lowercase_cow(&self) -> std::borrow::Cow<Self>;
-    /// Returns the same value as String::to_lowercase. The only difference
-    /// is that this functions returns ```Cow``` and does not allocate
-    /// if the string is already in lowercase.
-    fn to_ascii_lowercase_cow(&self) -> std::borrow::Cow<Self>;
+    fn to_lowercase_cow(&self) -> Cow<Self>;
 }
 
-impl StrOnlyExtension for str {
+impl StrLikeExtension for str {
```

Comment:
Why did we rename the type? It seems unnecessary for the scope of the PR

Reply:
I didn't rename the type, I just removed `to_ascii_lowercase_cow` from `StrOnlyExtension` because it is already present in `StrLikeExtension`. Now `str` implements both `StrLikeExtension` and `StrOnlyExtension`. This avoids duplicating all methods from `StrLikeExtension` in `StrOnlyExtension`.

Reply:
Thank you for the explanation, I just wanted to know if they were necessary or not, in order to reduce the diffs and make the PR easier to review. No worries.